src/path_planner_node.cpp

```diff
@@ -90,12 +91,22 @@ void PathPlanner::update_plan()
 
     RCLCPP_INFO(this->get_logger(), "Updating plan.");
 
+    /*
     auto bfs = BreadthFirstSearch(*last_map_state_);
     nav_msgs::msg::Path bfs_path = bfs.get_path(*last_pos_state_, last_goal_state_->pose);
     
     bfs_path.header.frame_id = map_frame_;
     bfs_path.header.stamp = this->get_clock()->now();
     path_publisher_->publish(bfs_path);
+    */
+
+    auto rrt = RRT(*last_map_state_);
```

Comment:
Creating a new RRT for every update seems really expensive, cause it copies the entire costmap each time.

IMO we should make a path finder interface that is implimented by each planner, then store a unique_ptr to that interface in the planner node class. We use polymorphism to instantiate the interface with RRT like
```cpp
std::unique_ptr<PathFinder> pfinder = std::make_unique<RRT>();
```

I would then have RRT's map_ feild be a shared_ptr to Occupancy grid, rather than a value like it is now. This means that this line here would become something like `rrt.update_map(last_map_state_)`, which would clone the smart pointer but not copy the occupancy grid behind it, saving a ton of time (I should note that last_map_state_ is actually already a smart pointer from ROS for this reason).